pkg/cache/clusterscoped_cache.go

```diff
@@ -0,0 +1,274 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+	toolscache "k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// a new global cluster cache to handle cluster scoped resources.
+const globalClusterCache = "_cluster"
```

Comment:
We don't need a `globalClusterCache` too eventually. Instead of having a separate global cache, we can iterate over all the cluster caches and return objects from each one of them.

Reply:
hmm, we'd still need a cache on `*` because multi-cluster requests would go there and could contain information not yet included in any specific cluster's cache, though we could just treat `*` like any other cache since it's not keyed differently. I guess the only other open question here is what we do about the top level admin-cluster scope - we may eventually want to set a cache on `[::1]:6443` in addition to all the `[::1]:6443/clusters/{cluster}` caches

Reply:
The `*` cluster scope is only to list and watch, so we can iterate over all the other caches to get that info. What did you mean by, `information not yet included in any specific cluster's cache` ?